app/models/agent_builder.rb

```diff
@@ -0,0 +1,18 @@
+class AgentBuilder
+  class << self
+    def build_for_type(type, user, attributes = {})
+      begin
+        klass = Agents.const_get(type)
```

Comment:
I think that `valid_type?(type)` was a way to avoid calling either `const_get` or `constantize` on something coming from a user.  Is that worth keeping, or is this safe?